PFOffer.sol

```diff
@@ -98,6 +118,7 @@ contract PFOffer {
         oneTimeCosts = _oneTimeCosts;
         minDailyWithdrawLimit = _minDailyWithdrawLimit;
         dailyWithdrawLimit = _minDailyWithdrawLimit;
+        proposalID = _proposalID;
```

Comment:
Check how I implemented the watch function to prevent that:

https://gist.github.com/alexvandesande/122c05a0f8a7108523aed29ae7ec92b1#file-proposal-js-L271

It checks if the address is the same and if the proposal has not been voted
yet. Someone could still trick it by submitting two proposals, but then the
token holders would be able to see that.

On Tue, Jun 7, 2016 at 9:02 AM, Lefteris Karapetsas <
notifications@github.com> wrote:

> In PFOffer.sol
> <https://github.com/slockit/DAO/pull/217#discussion_r66057366>:
>
> > @@ -98,6 +118,7 @@ contract PFOffer {
> >          oneTimeCosts = _oneTimeCosts;
> >          minDailyWithdrawLimit = _minDailyWithdrawLimit;
> >          dailyWithdrawLimit = _minDailyWithdrawLimit;
> > +        proposalID = _proposalID;
>
> Yes I have considered that. It's easy to get the proposal ID. It's just
> always proposalsNumber()+1.
>
> The problem with this approach is that you gotta make the proposal right
> after you deploy the contract. And even when you do you may collide with
> someone else who is making a proposal at the same time.
>
> I had not thought of a separate watchContract function. That's a nice
> idea. But we need to have a way to prevent the contractor from setting it
> to a proposalID that's not theirs in order to fool the checkVoteStatus()
> function.
>
> Probably the easiest way to do so is by means of community awareness. If
> the proposalID does not match the one that actually tries to sign() the
> contract we can put out a community alert that the contractor is a scammer.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/slockit/DAO/pull/217/files/5620e943d777294e89ca7e1263df220847a10a92#r66057366>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAG5AtCRRr7jH2XcVM5fm5YwwEVdi_LYks5qJV3rgaJpZM4IvUqK>
> .
>



-- 
Alex Van de Sande
Lead Designer
Ethereum Foundation